agent/client/client.go

```diff
@@ -220,6 +222,40 @@ func (c *Client) Run(ctx context.Context) error {
 	return nil
 }
 
+// GetAgentsStatus sends status of running agents to server
+func (c *Client) GetAgentsStatus() {
+	// Drain all existing messages
+loop:
+	for {
+		select {
+		case _, ok := <-c.supervisor.Changes():
+			if !ok {
+				break loop
+			}
+		default:
+			break loop
+		}
+	}
```

Comment:
I would extract this logic to a separate method, cleaning the `changes` channel seems out of responsibility of `GetAgentsStatus` method and instead of `break` we will be able to use just `return`. I would even move it inside `supervisor` and call it before calling `GetAgentsStatus` method.
BTW `GetAgentsStatus` seems like incorrect naming, maybe `SendActualStatuses`?

Reply:
don't see changes